pkg/render/amazoncloudintegration.go

```diff
@@ -0,0 +1,313 @@
+// Copyright (c) 2019 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package render
+
+import (
+	"fmt"
+	"strings"
+
+	operator "github.com/tigera/operator/pkg/apis/operator/v1"
+	"github.com/tigera/operator/pkg/components"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	AmazonCloudIntegrationNamespace      = "tigera-amazon-cloud-integration"
+	AmazonCloudIntegrationComponentName  = "tigera-amazon-cloud-integration"
+	AmazonCloudIntegrationCredentialName = "amazon-cloud-integration-credentials"
+	AmazonCloudCredentialKeyIdName       = "key-id"
+	AmazonCloudCredentialKeySecretName   = "key-secret"
+)
+
+func AmazonCloudIntegration(aci *operator.AmazonCloudIntegration, installation *operator.Installation, cred *AmazonCredential, ps []*corev1.Secret, openshift bool) (Component, error) {
+	return &amazonCloudIntegrationComponent{
+		amazonCloudInt: aci,
+		installation:   installation,
+		credentials:    cred,
+		pullSecrets:    ps,
+		openshift:      openshift,
+	}, nil
+}
+
+type amazonCloudIntegrationComponent struct {
+	amazonCloudInt *operator.AmazonCloudIntegration
+	installation   *operator.Installation
+	credentials    *AmazonCredential
+	pullSecrets    []*corev1.Secret
+	openshift      bool
```

Comment:
I wonder if we can leave this out and hardcode to false when calling `createNamespace`

Reply:
We could do an OpenShift install on AWS and add the cloud controller. I think we'd need this. (though it sounds like we may not need the thing done in createNamespace for openshift installs) I think I'd prefer to leave this for now.